prometheo/datasets/scaleag.py

```diff
@@ -129,6 +129,8 @@ def __init__(
             )
 
     def set_num_outputs(self) -> Optional[int]:
+        from loguru import logger  # Localized import
```

Comment:
I think we discussed this already but what is the advantage of a localized import here?  It seems like we use `loguru` throughout the codebase now (e.g. in `utils.py`)?  IIRC it was so that we could run the inference code without needing to install `loguru`.

I wonder if the benefits of loguru is worth these local imports (and also hidden dependencies). I would prefer to just use `logging`